lib/date.nit

```diff
@@ -0,0 +1,254 @@
+		year = y
+		month = mth
+		days = d
+		hours = h
+		minutes = m
+		sec = s
+	end
+
+	redef init today do
+		super
+		var tm = new Tm.localtime
+		hours = tm.hour
+		minutes = tm.min
+		sec = tm.sec
+	end
+
+	redef fun ==(d)  
+	do
+		if d isa DateTime and self.diff_days(d) == 0 then 
+			return time_eq(d)
+		end
+		return false
+	end
+
+	redef fun to_s do
+		super
+		return date_s + " {hours.to_s}:{minutes.to_s}:{sec.to_s}{time_zone}"
+	end
+end
+
+# The class "Date" allow the manipulation only on the date data
+class Date
+	super Comparable
+
+	# Years as an integer ex: 1989
+	var year: Int = 0
+
+	# Months as an integer ex: "3" for March
+	var month: Int = 0
+
+	# Days as an integer 
+	var days: Int = 0
+
+	# UTC time zone
+	var time_zone = "Z"
+
+	# Date : String format
+	var date_s: String = ""
+
+	# this will sets the date
+	init set(y, m, d: Int) do
+		year = y
+		month = m
+		days = d
+	end
+
+	# this will initialize the current date and time
+	init today do
+
+		var tm = new Tm.localtime
+		year = 1900 + tm.year
+		month = tm.mon + 1
+		days = tm.mday
+	end
+
+	# date and time expressed according to ISO 8601
+	redef fun to_s 
+	do
+		var y = year.to_s
+		var mth = month.to_s
+		var d = days.to_s
+
+		date_s = "{y}-{mth}-{d}"
+		return date_s
+	end
+	
+	# Compare two Date and return the difference in days
+	fun diff_days(d2: Date): Int 
+	do
+		var y_out  = 0
+		var m_out  = 0
+		var out = 0 
+
+		y_out = year - d2.year
+		y_out = y_out * 365
+		m_out = month - d2.month
+		m_out = m_out * 30
+		out = days - d2.days + m_out + y_out
+
+		return out
+	end
+
+	# Compare two date and return the difference in months
+	fun diff_month(d2: Date): Int
+	do
+		var y_out = 0
+		var out = 0
+
+		y_out = year - d2.year
+		y_out = y_out * 12
+		out = month - d2.month + y_out
+
+		return out
+	end
+
+	# Compare two date and return the difference in years
+	fun diff_years(d2: Date): Int
+	do
+		return year - d2.year
+	end
+
+	# Is self equals to d ?
+	redef fun ==(d)  
+	do
+		if d isa Date and self.diff_days(d) == 0 then
+			return true
+		end
+		return false
+	end
+
+	# Compare two date
+	redef fun <(d)
+	do
+		return d isa Date and self.diff_days(d) < 0
+	end
+
+	redef fun hash 
+	do
+		# the hash function must be review
+		return year + month * 1024 + days * 2048
+	end
+
+	# Check if the date(self) is between the interval (for years)
+	fun is_between_years(d1, d2: Date): Bool 
+	do
+		return (d1.year > year and d2.year < year) or (d2.year > year and d1.year < year) or (d1.year == year or d2.year == year)
+	end
+
+	# Check if the date(self) is between the interval (for month)
+	fun is_between_month(d1, d2: Date) : Bool
+	do
+		if not self.is_between_years(d1,d2) then return false
+		return (d1.month > month and d2.month < month) or (d2.month > month and d1.month < month) or (d1.month == month or d2.month == month)
+	end
+
+	# Check if the date(self) is between the interval (for days, the time is not supported for now)
```

Comment:
I think i'll change replace this by : the current date instance ? (date(self) == the class instance)